lib/Conversion/AffineToPipeline/AffineToPipeline.cpp

```diff
@@ -264,14 +270,46 @@ LogicalResult AffineToPipeline::populateOperatorTypes(
           problem.setLinkedOperatorType(combOp, combOpr);
           return WalkResult::advance();
         })
-        .Case<AffineLoadOp, AffineStoreOp, memref::LoadOp, memref::StoreOp>(
-            [&](Operation *seqOp) {
-              // Some known sequential ops. In certain cases, reads may be
-              // combinational in Calyx, but taking advantage of that is left as
-              // a future enhancement.
-              problem.setLinkedOperatorType(seqOp, seqOpr);
-              return WalkResult::advance();
-            })
+        .Case<AddIOp, CmpIOp>([&](Operation *seqOp) {
+          // These ops need to be sequential for now because we do not
+          // have enough information to chain them together yet.
+          problem.setLinkedOperatorType(seqOp, seqOpr);
+          return WalkResult::advance();
+        })
+        .Case<AffineStoreOp, memref::StoreOp>([&](Operation *memOp) {
+          // Some known sequential ops. In certain cases, reads may be
+          // combinational in Calyx, but taking advantage of that is left as
+          // a future enhancement.
+          TypedValue<MemRefType> memRef =
+              isa<AffineStoreOp>(*memOp)
+                  ? cast<AffineStoreOp>(*memOp).getMemRef()
+                  : cast<memref::StoreOp>(*memOp).getMemRef();
+          memOps[&memRef] =
+              std::pair(memOps[&memRef].first, memOps[&memRef].second + 1);
+          Problem::OperatorType memOpr = problem.getOrInsertOperatorType(
+              "mem_" + std::to_string(hash_value(memRef)));
+          problem.setLatency(memOpr, 1);
+          problem.setLimit(memOpr, 1);
+          problem.setLinkedOperatorType(memOp, memOpr);
+          return WalkResult::advance();
+        })
+        .Case<AffineLoadOp, memref::LoadOp>([&](Operation *memOp) {
+          // Some known sequential ops. In certain cases, reads may be
+          // combinational in Calyx, but taking advantage of that is left as
+          // a future enhancement.
+          TypedValue<MemRefType> memRef =
+              isa<AffineLoadOp>(*memOp)
+                  ? cast<AffineLoadOp>(*memOp).getMemRef()
+                  : cast<memref::LoadOp>(*memOp).getMemRef();
+          memOps[&memRef] =
+              std::pair(memOps[&memRef].first + 1, memOps[&memRef].second);
+          Problem::OperatorType memOpr = problem.getOrInsertOperatorType(
+              "mem_" + std::to_string(hash_value(memRef)));
```

Comment:
So this, combined with setting limit to 1 means you intend each memref to represent a unique memory resource on the physical hardware?